core/src/main/kotlin/org/evomaster/core/problem/rest/service/RestResourceMutator.kt

```diff
@@ -48,9 +49,9 @@ class ResourceRestMutator : StandardMutator<RestIndividual>() {
         return individual.getResourceCalls().flatMap { it.seeGenes() }.filter(Gene::isMutable)
     }
 
-    override fun update(previous: EvaluatedIndividual<RestIndividual>, mutated: EvaluatedIndividual<RestIndividual>, mutatedGenes : MutableList<Gene>) {
+    override fun update(previous: EvaluatedIndividual<RestIndividual>, mutated: EvaluatedIndividual<RestIndividual>, mutatedGenes : MutatedGeneSpecification?) {
         //update resource dependency after mutating structure of the resource-based individual
-        if(mutatedGenes.isEmpty() && (previous.individual.getResourceCalls().size > 1 || mutated.individual.getResourceCalls().size > 1) && config.probOfEnablingResourceDependencyHeuristics > 0){
+        if(mutatedGenes!!.mutatedGenes.isEmpty() && (previous.individual.getResourceCalls().size > 1 || mutated.individual.getResourceCalls().size > 1) && config.probOfEnablingResourceDependencyHeuristics > 0){
```

Comment:
don't understand this: in the function input, you marked it nullable with "?", but here you say it is never null with "!!"

Reply:
RestResourceMutator can be only applied with MIO. In MIO, the input is not null by default.

Reply:
in that case, can you add such comment there in the code?